nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentJsonReader.java

```diff
@@ -59,27 +60,23 @@ protected Comment readEntity(JsonNode jn) {
 
     protected static Comment fillCommentEntity(JsonNode jn, Comment comment) {
         // don't read id from given JsonNode, if needed it is read from path
-        comment.setParentId(jn.get(COMMENT_PARENT_ID_FIELD).textValue());
-        comment.setText(jn.get(COMMENT_TEXT_FIELD).textValue());
-
-        JsonNode creationDateNode = jn.get(COMMENT_CREATION_DATE_FIELD);
-        Instant creationDate = creationDateNode != null && !creationDateNode.isNull()
-                ? Instant.parse(creationDateNode.textValue())
-                : null;
-        comment.setCreationDate(creationDate);
-
-        JsonNode modificationDateNode = jn.get(COMMENT_MODIFICATION_DATE_FIELD);
-        Instant modificationDate = modificationDateNode != null && !modificationDateNode.isNull()
-                ? Instant.parse(modificationDateNode.textValue())
-                : null;
-        comment.setModificationDate(modificationDate);
-
-        if (jn.has(EXTERNAL_ENTITY_ID_FIELD)) {
+        optionalSet(jn, COMMENT_PARENT_ID_FIELD, comment::setParentId);
+        optionalSet(jn, COMMENT_TEXT_FIELD, comment::setText);
+        optionalSet(jn, COMMENT_CREATION_DATE_FIELD, s -> comment.setCreationDate(s == null ? null : Instant.parse(s)));
+        optionalSet(jn, COMMENT_MODIFICATION_DATE_FIELD,
```

Comment:
i think `optionalSet(jn, COMMENT_CREATION_DATE_FIELD, s -> comment.setCreationDate(s == null ? null : Instant.parse(s)));` can be replaced by `optionalSet(jn, COMMENT_CREATION_DATE_FIELD, s -> comment.setCreationDate(Instant.parse(s)));` as the `optionalSet` checks the `jn.has(key)` which 
```
 public boolean has(String fieldName) {
        return this.get(fieldName) != null;
    }
```

and i am wondering if we can find / replace the method name  it makes us think to the `java.util.Optional` -> `setIfExists`, `setIfDefined`

Reply:
But `s` can be `null`. `this.get(fieldName)` returns a `JsonNode` that won't be `null` if the key exist, but its value can be `null`.

Reply:
Indeed @troger, that's what we found in the javadoc of the method you pasted @RSalem07 
```java
    /**
     * Method that allows checking whether this node is JSON Object node
     * and contains value for specified property. If this is the case
     * (including properties with explicit null values), returns true;
     * otherwise returns false.
     *<p>
     * This method is equivalent to:
     *<pre>
     *   node.get(fieldName) != null
     *</pre>
     * (since return value of get() is node, not value node contains)
     *<p>
     * NOTE: when explicit <code>null</code> values are added, this
     * method will return <code>true</code> for such properties.
     *
     * @param fieldName Name of element to check
     * 
     * @return True if this node is a JSON Object node, and has a property
     *   entry with specified name (with any value, including null value)
     */
    public boolean has(String fieldName) {
        return get(fieldName) != null;
    }
```

Reply:
yep i checked too after the thomas comment

Reply:
@kevinleturc and what the name:

>and i am wondering if we can find / replace the method name it makes us think to the java.util.Optional -> setIfExists, setIfDefined

i mean perhaps replace it by `setIfExists` ...

Reply:
Was thinking about it, yes why not for `setIfExists`